src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -15001,35 +15001,204 @@ def stack(
+
+        Returns a DataFrame having a new level of column labels whose
+        inner-most level consists of the pivoted index labels.
+
+        If the index is not a MultiIndex, the output will be a Series
+        (the analogue of stack when the columns are not a MultiIndex).
+
+        Parameters
+        ----------
+        level : int, str, list, default -1
+            Level(s) of index to unstack, can pass level name.
+
+        fillna : int, str, dict, optional
+            Replace NaN with this value if the unstack produces missing values.
+
+        sort : bool, default True
+            Sort the level(s) in the resulting MultiIndex columns.
+
+        is_series_output : bool, default False
+            Whether the output is a Series, used by `fillna`
+        """
+        if not isinstance(level, int):
+            ErrorMessage.not_implemented(
+                "Snowpark pandas DataFrame/Series.unstack does not yet support a non-integer `level` parameter"
+            )
+        if not sort:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas DataFrame/Series.unstack does not yet support the `sort` parameter"
+            )
+        level = level if is_list_like(level) else [level]
+
+        index_names = self.get_index_names()
+        has_index_names_none = None in index_names
+        # Check to see if we have a MultiIndex, if we do, make sure we remove
+        # the appropriate level(s), and we pivot accordingly.
+        if len(index_names) > 1:
+            if has_index_names_none:
+                # Index name defaults to "level_{i}" after reset_index() operation
+                # if index name is None
+                index_cols = [
+                    f"level_{i}" if index_names[i] is None else index_names[i]
+                    for i in range(len(index_names))
+                ]
+            else:
+                index_cols = index_names  # type: ignore
+
+            pivot_cols = [index_cols[lev] for lev in level]  # type: ignore
+            index_cols = [
+                index_cols[i]
+                for i in range(len(index_cols))
+                if index_cols[i] not in pivot_cols
+            ]
+            vals = [
+                c for c in self.columns if c not in index_cols and c not in pivot_cols
+            ]
+            qc = self.reset_index().pivot_table(
+                columns=pivot_cols,
+                index=index_cols,
+                values=vals,
+                aggfunc="min",
+                fill_value=None,
+                margins=False,
+                dropna=True,
+                margins_name="All",
+                observed=False,
+                sort=sort,
+            )
+
+            # Set the correct index names based on if index name is None
+            output_index_names = qc.get_index_names()
+            output_index_names_replace_level_with_none = [
+                None
+                if isinstance(output_index_names[i], str)
```

Comment:
here it seems because the None columns is set to other names, and here we are trying to reset it back. However, those kind of name based operation could be error prone, since level_* is a pretty common used default name in pandas, and the dataframe feed into this operation could be a dataframe that contains those labels already.

I think we can keep the original index names and columns names, and derive the final name  based on the original name, probably with positions, order is something will be maintained during the operations.

Also, we can simply recreate the internal frame here to update the names, which seems much  simplier to do here